auth.ts

```diff
@@ -1,127 +1,144 @@
-require('dotenv').config({ silent: true });
-
-import jwksClient from 'jwks-rsa';
-import jwt from 'jsonwebtoken';
-import util from 'util';
-
-import { APIGatewayProxyEvent } from 'aws-lambda';
-
-export default class Auth implements IAuth {
-  constructor() {}
-
-  private async getSigningKey(keyId: string): Promise<string> {
-    console.log('testing', keyId);
-    const client = jwksClient({
-      cache: true,
-      rateLimit: true,
-      jwksRequestsPerMinute: 10,
-      jwksUri: process.env.JWKS_URI || ''
-    });
-
-    console.log('testing####', client);
-
-    const retrieveSigningKey = util.promisify(client.getSigningKey);
-    console.log('retrieveSigningKey####', retrieveSigningKey);
-    const retrievedKey = await retrieveSigningKey(keyId);
-    console.log('retrievedKey####', retrievedKey);
-    return (retrievedKey as jwksClient.CertSigningKey).publicKey ||
-    (retrievedKey as jwksClient.RsaSigningKey).rsaPublicKey;
+import jwksClient from 'jwks-rsa'
+import jwt from 'jsonwebtoken'
+import util from 'util'
+
+import { APIGatewayProxyEvent } from 'aws-lambda'
+
+export const createCheckScopesAndResolve = ({
+  jwksUri,
+  audience,
+  issuer,
+}: {
+  jwksUri: string
+  audience: string
+  issuer: string
+}) => {
+  const client = jwksClient({
+    cache: true,
+    rateLimit: true,
+    jwksRequestsPerMinute: 10,
+    jwksUri,
+  })
+
+  const retrieveSigningKey = util.promisify(client.getSigningKey)
+
+  const getSigningKey = async (keyId: string): Promise<string> => {
+    const retrievedKey = await retrieveSigningKey(keyId)
+    return (
+      (retrievedKey as jwksClient.CertSigningKey).publicKey ||
+      (retrievedKey as jwksClient.RsaSigningKey).rsaPublicKey
+    )
   }
 
-  private extractBearerToken(event: APIGatewayProxyEvent): string {
```

Comment:
This member method does not use the state of the class instance once (no access to `this`) so it should either but `static` or even declared outside of the class, for example as a sole function at the end of the file.